code/game/objects/items/devices/flashlight.dm

```diff
@@ -312,17 +312,20 @@ obj/item/device/flashlight/lamp/bananalamp
 		..()
 	return
 
-/obj/item/device/flashlight/emp/afterattack(atom/A as mob|obj, mob/user, proximity)
-	if(!proximity) return
-	if(istype(A, /obj/item/weapon/storage/) && A.loc == user)
+/obj/item/device/flashlight/emp/afterattack(atom/movable/A, mob/user, proximity)
+	if(!proximity)
 		return
-	if (emp_cur_charges > 0)
+
+	if(emp_cur_charges > 0)
 		emp_cur_charges -= 1
-		A.visible_message("<span class='danger'>[user] blinks \the [src] at \the [A].", \
-											"<span class='userdanger'>[user] blinks \the [src] at \the [A].")

```

Comment:
the visible_message doesn't use the same argument when called by a mob or an object, causing issues here.